src/attestedclaim/AttestedClaim.spec.ts

```diff
@@ -99,4 +132,37 @@ describe('RequestForAttestation', () => {
     delete falsePresentation.request.claimHashTree[propertyName]
     expect(falsePresentation.verifyData()).toBeFalsy()
   })
+
+  it('compresses and decompresses the attested claims object', () => {
+    expect(compressAttestedClaim(legitimation)).toEqual(compressedLegitimation)
+
+    expect(decompressAttestedClaim(compressedLegitimation)).toEqual(
+      legitimation
+    )
+
+    expect(legitimation.compress()).toEqual(compressAttestedClaim(legitimation))
+
+    expect(AttestedClaim.decompress(compressedLegitimation)).toEqual(
+      legitimation
+    )
+  })
+
+  it('Negative test for compresses and decompresses the attested claims object', () => {
+    compressedLegitimation.pop()
+    delete legitimation.attestation
```

Comment:
I have changed the data structure of the `compressedLegitimation` and `legitimation` so, I can see if they throw and error if incorrect.